rdt-open-reader/src/main/java/io/ona/rdt/util/Utils.java

```diff
@@ -225,4 +229,48 @@ public static boolean isValidJSONObject(String str) {
             return false;
         }
     }
+
+    public static void verifyUserAuthorization(SyncUtils syncUtils) {
+        UserAuthorizationVerificationTask userAuthorizationVerificationTask = UserAuthorizationVerificationTask.getInstance(syncUtils);
+        if (userAuthorizationVerificationTask.getStatus().equals(AsyncTask.Status.PENDING)) {
+            userAuthorizationVerificationTask.execute();
+        } else if (userAuthorizationVerificationTask.getStatus().equals(AsyncTask.Status.FINISHED)) {
+            userAuthorizationVerificationTask.destroy();
+            verifyUserAuthorization(syncUtils);
+        }
+    }
+
+    public static class UserAuthorizationVerificationTask extends AsyncTask<Void, Void, Void> {
+
+        private static UserAuthorizationVerificationTask INSTANCE;
+        private final SyncUtils syncUtils;
+
+        public static UserAuthorizationVerificationTask getInstance(SyncUtils syncUtils) {
+            if (INSTANCE == null) {
+                INSTANCE = new UserAuthorizationVerificationTask(syncUtils);
+            }
+            return INSTANCE;
+        }
+
+        public UserAuthorizationVerificationTask(SyncUtils syncUtils) {
```

Comment:
This should be a private method in singleton pattern.

Reply:
done